### ISSUE TITLE
Something broke in `rclone serve restic` with docker image tag 1.61

### ISSUE THREAD

**Issue by saikpr** [rclone/rclone#6644]
#### The associated forum post URL from `https://forum.rclone.org`

Didn't post in forum as this was a regression.

#### What is the problem you are having with rclone?
TL;DR
Having issues with using `rclone serve` with `1.61`, didn't have issue with `1.60`.

Long story:

I use `rclone serve` to host a repo, here's my docker config - 

```
version: "2.4"
services:
  restic_server:
    image: rclone/rclone:latest
    restart: unless-stopped
    networks:
      - default
    container_name: restic_server
    command: serve restic -vvv --fast-list --transfers 16 restic-backup-storage:restic-backup-storage --append-only --baseurl XXX --addr :8080 --htpasswd /config/rclone/htpasswd
    volumes:
      - ./rclone_creds:/config/rclone
    ports:
      - 127.0.0.1:16080:8080/tcp
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/Los_Angeles

networks:
  default:
```

but recently this container started to fail after it was updated to 1.61 tag [docker repo](https://hub.docker.com/layers/rclone/rclone/1.61/images/sha256-6dfc66aa9eb36ee5ce00a7d646726949a7182ec6b5545e915aaf91e8e1329694?context=explore), but it still works with 1.60 tag [docker repo](https://hub.docker.com/layers/rclone/rclone/1.60/images/sha256-8c7a6492379247c606401281412d865207c4584d959f7530c1cff3e98172e4f1?context=explore).

Here's my docker version as well (if it helps)

```
sainyam@app-az:~/apps/restic-rclone-server$ docker version
Client: Docker Engine - Community
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 18:01:19 2022
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       3056208
  Built:            Tue Oct 25 17:59:41 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.10
  GitCommit:        770bd0108c32f3fb5c73ae1264f7e503fe7b2661
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
  ```
  

#### What is your rclone version (output from `rclone version`)

Doesn't work on 1.61 but works on 1.60
```
docker run rclone/rclone:1.61 version
rclone v1.61.0
- os/version: alpine 3.17.0 (64 bit)
- os/kernel: 5.10.0-19-arm64 (aarch64)
- os/type: linux
- os/arch: arm64
- go/version: go1.19.4
- go/linking: static
- go/tags: none

```

```
docker run rclone/rclone:1.60 version
rclone v1.60.1
- os/version: alpine 3.16.3 (64 bit)
- os/kernel: 5.10.0-19-arm64 (aarch64)
- os/type: linux
- os/arch: arm64
- go/version: go1.19.3
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Debian 11

#### Which cloud storage system are you using? (e.g. Google Drive)
Backblaze b2

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

See docker-compose above

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
Attaching to restic_server
restic_server  | 2022/12/20 22:57:41 DEBUG : rclone: Version "v1.61.0" starting with parameters ["rclone" "serve" "restic" "-vvv" "--fast-list" "--transfers" "16" "restic-backup-storage:restic-backup-storage" "--append-only" "--baseurl" "my-restic-server" "--addr" ":8080" "--htpasswd" "/config/rclone/htpasswd"]
restic_server  | 2022/12/20 22:57:41 DEBUG : Creating backend with remote "restic-backup-storage:restic-backup-storage"
restic_server  | 2022/12/20 22:57:41 DEBUG : Using config file from "/config/rclone/rclone.conf"
restic_server  | 2022/12/20 22:57:41 INFO  : Using "/config/rclone/htpasswd" as htpasswd storage
restic_server  | 2022/12/20 22:57:41 NOTICE: B2 bucket restic-backup-storage: Serving restic REST API on [http://[::]:8080/my-restic-server/]
restic_server  | 2022/12/20 22:57:41 INFO  :
restic_server  | Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
restic_server  | Elapsed time:         0.3s
restic_server  |
restic_server  | 2022/12/20 22:57:41 DEBUG : 5 go routines active
restic_server exited with code 0
restic_server  | 2022/12/20 22:57:41 DEBUG : rclone: Version "v1.61.0" starting with parameters ["rclone" "serve" "restic" "-vvv" "--fast-list" "--transfers" "16" "restic-backup-storage:restic-backup-storage" "--append-only" "--baseurl" "my-restic-server" "--addr" ":8080" "--htpasswd" "/config/rclone/htpasswd"]
restic_server  | 2022/12/20 22:57:41 DEBUG : Creating backend with remote "restic-backup-storage:restic-backup-storage"
restic_server  | 2022/12/20 22:57:41 DEBUG : Using config file from "/config/rclone/rclone.conf"
restic_server  | 2022/12/20 22:57:42 INFO  : Using "/config/rclone/htpasswd" as htpasswd storage
restic_server  | 2022/12/20 22:57:42 NOTICE: B2 bucket restic-backup-storage: Serving restic REST API on [http://[::]:8080/my-restic-server/]
restic_server  | 2022/12/20 22:57:42 INFO  :
restic_server  | Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
restic_server  | Elapsed time:         0.2s
restic_server  |
restic_server  | 2022/12/20 22:57:42 DEBUG : 5 go routines active
restic_server exited with code 0
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
`restic serve` got refactored and I expect what you are seeing is fallout from that.

It does look broken - I agree. I don't know how it passed the unit and integration tests...

Try this

[v1.62.0-beta.6657.bfd225166.fix-6644-serve-restic](https://beta.rclone.org/branch/fix-6644-serve-restic/v1.62.0-beta.6657.bfd225166.fix-6644-serve-restic/) on branch [fix-6644-serve-restic](https://github.com/rclone/rclone/tree/fix-6644-serve-restic) (uploaded in 15-30 mins)

That doesn't build a docker image - I suggest you log into your docker image and replace the rclone binary temporarily with the right one from there to see if it works.

If it works I'll push it to master which will build a docker image.

**Comment by ncw** [comment]
This fix has been released in v1.61.1 which is available now

**Comment by saikpr** [comment]
Hey Nick, Thanks for taking a look into this and quickly fixing it! I cannot tell you how much useful rclone has been for me!
